### ISSUE TITLE
Make Wifi only mode passion on iOS for 3 locally

### ISSUE THREAD

**Issue by larryonoff** [thaliproject/Thali_CordovaPlugin#1029]

**Comment by dersim-davaod** [comment]
Today we tried to run tests on 3 devices locally in Minsk.
For now some of them fails. For example <code>Client to server request coordinated</code>.
@artemjackson will investigate this issue tomorrow.

I will disable invalid test files and will try to run tests again, so we will have a set of problem files by tomorrow.

**Comment by larryonoff** [comment]
@artemjackson tested wifi-only mode on 3 iOS devices with @dersim-davaod, issue #1029.

He faced with an issue with `Socket ping timeout` on `Client to server request coordinated` test, so currently not all tests passed on iOS. We need some time to investigate it deeper. Will work on it tomorrow.

@artemjackson tried to fix 'superbranch' again (in #1010). But logs indicate that something is wrong with teardown, i.e. `# 15. #_doImmediateSeqUpdate - create new seq doc`. So @aaladev tried to fix teadrown issue in #1007. @artemjackson created and worked on a new ticket #1013 where he was trying to fix the ’superbranch’.

@artemjackson reviewed #1007 (@aaladev 's  test Server) and merged it with ’superbrach’ in a separate branch (#1010), but currently it fails on build. Will investigate in deeper tomorrow.

**Comment by larryonoff** [comment]
@artemjackson are we having issues with new test server or the old one? Why we cannot try the old one server in case if we're having issues with new test server?

**Comment by artemjackson** [comment]
@larryonoff we have the teardown issue with old test server, @andrew-aladev fixing the issue by completly rewriting the server.
I tried to use the @andrew-aladev's test server implementation by merging #1007 into #1010 but got a build failure. Tomorrow I'll investgate and fix it.

**Comment by artemjackson** [comment]
#1010 moved to #1033

**Comment by artemjackson** [comment]
#1010 moved to #1033

**Comment by dersim-davaod** [comment]
@artemjackson,
for now we have following results:

we definitely have an issue with:
 1. `Client to server request coordinated` test in `testThaliNotification.js` file.
 2. `Coordinated seq test` in `testLocalSeqManagerCoordinated.js`
 3. `Coordinated replication action test` in `testThaliReplicationPeerActionCoordinated.js`
 4. `test write` in `testThaliManagerCoordinated.js`
 5. `test thali manager spies`
 6. `Coordinated pull replication from notification test` in  `testThaliPullReplicationFromNotificationCoordinated.js`

Each of them fails every time I run it.

Also for now we have issues with following tests (they can fail randomly from time to time, so I think the real reason may be different):

1. `emits routerPortConnectionFailed`
    * output: `info: Socket to device iPhone disconnected: ping timeout`
2. `Resolves an action locally`
    * output: `debug: Too many emit retries to device: iPhone`
3. `does not emit duplicate discoveryAdvertisingStateUpdate`
    * output: `debug: Too many emit retries to device: iPhone`
4. `should be able to call #startListeningForAdvertisements many times`
    * output: `info: Socket to device iPhone disconnected: ping timeout`
5. `Do nothing and make sure we time out`
    * output: `debug: Too many emit retries to device: iPhone`
6. `Make sure docs replicate`
    * output: `info: Socket to device iPhone disconnected: transport error`
7. `#update - do we cancel timer properly on an immediate?`
    * output: `warn: Failed on ios test: 24. #update - do we cancel timer properly on an immediate?`
8. `#update - test that we pick up new sequences while we wait`
9. `should be able to call #startListeningForAdvertisements many times`
    * output: `info: Socket to device iPhone disconnected: ping timeout`


On the other hand when I moved invalid files into "disabled" folder, then all other tests passed locally (see full log with 3 iOS devices  [here](https://www.dropbox.com/s/9kzt6flwyu2o8sp/full_tests_log_3_devices.txt?dl=0))

**Comment by dersim-davaod** [comment]
Also following test fails randomly:
`does not emit duplicate discoveryAdvertisingStateUpdate`
output:
`info: Socket to device iPhone disconnected: ping timeout`
`debug: Too many emit retries to device: iPhone`

Then all tests passed (log [here](https://www.dropbox.com/s/a6qk2qdaelpun7t/full_tests_log_3_devices_14sep_evening.txt?dl=0))

**Comment by yaronyg** [comment]
1. Get 3 Android devices passing all wifi only tests with 100% reliability in Minsk
2. Get 3 iOS devices passing all wifi only test with 100% reliability in Minsk

**Comment by artemjackson** [comment]
@chapko got 3 Android devices passing 3/3 times in a row with all tests enabled from 'superbranch'
@evabishchevich got 3 Android devices passing 2/3 times with all tests enabled from #1033 (it's a 'superbranch' with `setImmediate` workaround without which CI devices didn't work).
At one of the test runs he faced the issue #1057

**Comment by artemjackson** [comment]
For iOS problem tests are:
`testThaliNotification.js`
`testLocalSeqManager.js`
`testLocalSeqManagerCoordinated.js`
`testThaliManagerCoordinated.js`
`testThaliReplicationPeerActionCoordinated.js`
`testThaliPullReplicationFromNotificationCoordinated.js`
`testCreateNativeListener.js` (what? it shouldn't)
`testThaliNotificationClient.js`
`testThaliMobile.js`
`testThaliManager.js`
`testThalireplicationPeerAction.js`

**Comment by yaronyg** [comment]
So you mean all the tests in all those files fail?

**Comment by yaronyg** [comment]
@dersim-davaod Once @andrew-aladev has checked in thaliproject/Thali_CordovaPlugin#1061 and thaliproject/Thali_CordovaPlugin#1064 and once he has finished and had PRs accepted for thaliproject/Thali_CordovaPlugin#1070, thaliproject/Thali_CordovaPlugin#1072 and thaliproject/Thali_CordovaPlugin#1073 THEN please re-run the iOS WiFi only tests. Please run all tests, see which ones fail and record them. We need to find one test that passes reliably on WiFi only mode on Android (ask @evabishchevich for help) but fails reliably on iOS. Then please assign the bug with the name of that test to me.

**Comment by dersim-davaod** [comment]
@artemjackson, log file with test results on clear `origin/vNext_artemjackson_1015`: [log](origin/vNext_artemjackson_1015)
We have `info: Socket to device Apple-iPhone disconnected: ping timeout` during `Coordinated seq test`.

**Comment by dersim-davaod** [comment]
Log file with tests running on simulators (the same branch `origin/vNext_artemjackson_1015`): [log](https://gist.github.com/dersim-davaod/a52ccceeb61c07e32e369f4a5e44ccd7).
We also have `info: Socket to device Apple-iPhone Simulator disconnected: ping timeout` during `Coordinated seq test`.

**Comment by artemjackson** [comment]
@yaronyg I updated the https://github.com/thaliproject/Thali_CordovaPlugin/issues/1029#issuecomment-247613603

**Comment by dersim-davaod** [comment]
Also a small note.
Looks like `Coordinated seq test` passed on 2 simulators, but not when 3.

**Comment by dersim-davaod** [comment]
@artemjackson has merged @andrew-aladev's changes into `vNext_artemjackson_1015` and I've tried to run tests again.
We have `INFO HttpServer: 'Socket to device name: 'Apple-iPhone' disconnected, reason: 'ping timeout''` during `DEBUG UnitTestFramework: '#run: 'test write''`.
See full log [here](https://gist.github.com/dersim-davaod/1af8279ad882b3c9b0237cba250e0705)

**Comment by dersim-davaod** [comment]
@yaronyg,
I and @artemjackson decided that `test repeat write 1` tests fails everytime we run it.
So I've created a new issue #1082 and assigned it to you.

**Comment by yaronyg** [comment]
This will now be investigated by Enrico and we'll bring this bug back to active when Enrico gets a result.

**Comment by dersim-davaod** [comment]
@artemjackson,

also a couple of logs with new test server: [this one](https://gist.github.com/dersim-davaod/d7a9db674e63f532b0c50f1ed05e48a5) and [this one](https://gist.github.com/dersim-davaod/314306d2d76054490569d09bfd736cef).
 `Coordinated seq test ` and `test thali manager multiple starts and stops ` failed respectively.

**Comment by yaronyg** [comment]
This blocked on thaliproject/Thali_CordovaPlugin#1082
